proto/cosmos/tx/v1beta1/service.proto

```diff
@@ -17,6 +17,32 @@ service Service {
     rpc GetTx(GetTxRequest) returns (GetTxResponse) {
         option (google.api.http).get = "/cosmos/tx/v1beta1/tx/{hash}";
     }
+
+    // TxsByEvents fetches txs by event.
+    rpc TxsByEvents(GetTxsEventRequest) returns (TxsByEventsResponse) {
+        option (google.api.http).get = "/cosmos/tx/v1beta1/txs";
+    }
+}
+
+// GetTxsEventRequest is the request type for the Service.TxsByEvents
+// RPC method.
+message GetTxsEventRequest {
+
+    // event is the transaction event type.
+    string event = 1;
+
+    // page is the page number
+    int64 page = 2;
+
+    // limit defines per page limit
+    int64 limit = 3;
+}
+
+// TxsByEventsResponse is the response type for the Service.TxsByEvents
+// RPC method.
+message TxsByEventsResponse {
+    // txs is the list of queried transactions.
+    repeated cosmos.tx.v1beta1.Tx txs = 1;
```

Comment:
it would be great return more info about the txs too (height, gas used). Maybe we could create a `TxResponse` struct that looks exactly like abci's `TxResponse`, except that `string data = 5;` is `cosmos.tx.v1beta1.Tx tx = 5;`.

What do you think? (ideally, the txbyhash endpoint should also return this new TxResponse)

Reply:
yeah makes sense